components/file_preview/file_preview.tsx

```diff
@@ -1,38 +1,32 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactNode} from 'react';
 import {getFileThumbnailUrl, getFileUrl} from 'mattermost-redux/utils/file_utils';
 
 import FilenameOverlay from 'components/file_attachment/filename_overlay.jsx';
 import RemoveIcon from 'components/widgets/icons/fa_remove_icon';
 import Constants, {FileTypes} from 'utils/constants';
 import * as Utils from 'utils/utils.jsx';
 
-import FileProgressPreview from './file_progress_preview.jsx';
+import FileProgressPreview from './file_progress_preview';
 
-export default class FilePreview extends React.PureComponent {
-    static propTypes = {
-        enableSVGs: PropTypes.bool.isRequired,
-        onRemove: PropTypes.func.isRequired,
-        fileInfos: PropTypes.arrayOf(PropTypes.object).isRequired,
-        uploadsInProgress: PropTypes.array,
-        uploadsProgressPercent: PropTypes.object,
-    };
-
-    static defaultProps = {
-        fileInfos: [],
-        uploadsInProgress: [],
-        uploadsProgressPercent: {},
-    };
-
-    handleRemove = (id) => {
+type Props = {
+    enableSVGs: boolean;
+    onRemove: (id: string) => void;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    fileInfos: any[];
```

Comment:
Instead of using an `any` here, we can define a `FilePreviewInfo` type in this file for use by this file. It should contain whatever fields are currently consumed by the component.

Reply:
is there a list of fields defined somewhere? when I was looking it seemed to vary so I wasn't sure.

Reply:
You can define whatever the `file_preview.tsx` component consumes, and set the fields that are optional to nullable.